bndtools.core/src/org/bndtools/core/editors/ImportPackageQuickFixProcessor.java

```diff
@@ -0,0 +1,520 @@
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import org.bndtools.api.BndtoolsConstants;
+import org.bndtools.api.ILogger;
+import org.bndtools.api.Logger;
+import org.bndtools.core.ui.icons.Icons;
+import org.eclipse.core.resources.IFile;
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.resources.IResource;
+import org.eclipse.core.resources.IResourceRuleFactory;
+import org.eclipse.core.resources.IWorkspace;
+import org.eclipse.core.resources.ResourcesPlugin;
+import org.eclipse.core.resources.WorkspaceJob;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Status;
+import org.eclipse.jdt.core.ICompilationUnit;
+import org.eclipse.jdt.core.IJavaProject;
+import org.eclipse.jdt.core.compiler.IProblem;
+import org.eclipse.jdt.core.dom.ASTNode;
+import org.eclipse.jdt.core.dom.ImportDeclaration;
+import org.eclipse.jdt.core.dom.Name;
+import org.eclipse.jdt.core.dom.NameQualifiedType;
+import org.eclipse.jdt.core.dom.PackageDeclaration;
+import org.eclipse.jdt.core.dom.QualifiedName;
+import org.eclipse.jdt.core.dom.SimpleType;
+import org.eclipse.jdt.internal.corext.dom.ASTNodes;
+import org.eclipse.jdt.ui.text.java.IInvocationContext;
+import org.eclipse.jdt.ui.text.java.IJavaCompletionProposal;
+import org.eclipse.jdt.ui.text.java.IProblemLocation;
+import org.eclipse.jdt.ui.text.java.IQuickFixProcessor;
+import org.eclipse.jface.text.contentassist.IContextInformation;
+import org.eclipse.swt.graphics.Image;
+import org.eclipse.swt.graphics.Point;
+import org.osgi.framework.namespace.BundleNamespace;
+import org.osgi.framework.namespace.PackageNamespace;
+import org.osgi.resource.Capability;
+import org.osgi.resource.Requirement;
+import org.osgi.service.repository.Repository;
+
+import aQute.bnd.build.Project;
+import aQute.bnd.build.Workspace;
+import aQute.bnd.build.WorkspaceRepository;
+import aQute.bnd.build.model.BndEditModel;
+import aQute.bnd.build.model.clauses.VersionedClause;
+import aQute.bnd.header.Attrs;
+import aQute.bnd.osgi.resource.CapReqBuilder;
+import aQute.bnd.properties.Document;
+import aQute.bnd.properties.IDocument;
+import aQute.bnd.service.RepositoryPlugin;
+import aQute.lib.collections.MultiMap;
+import aQute.lib.io.ByteBufferInputStream;
+import aQute.lib.io.IO;
+import aQute.lib.strings.Strings;
+import bndtools.central.Central;
+import bndtools.central.RepositoryUtils;
+
+public class ImportPackageQuickFixProcessor implements IQuickFixProcessor {
+
+    private static final ILogger logger = Logger.getLogger(ImportPackageQuickFixProcessor.class);
+
+    // Relevance constants
+    final public static int ADD_BUNDLE = 15;
+    final public static int ADD_BUNDLE_WORKSPACE = ADD_BUNDLE + 1;
```

Comment:
I would prefer the visibility keyword to be first on these.

Reply:
accessibility keyword :-)

Reply:
Fixed. Any chance of this getting added to the code formatter so that it automatically puts all of the modifiers in the correct order (just like it auto-magically formats so much other stuff - whether I want it to or not? 😉 )